compiler/rustc_passes/src/loops.rs

```diff
@@ -213,15 +268,14 @@ impl<'a, 'hir> CheckLoopVisitor<'a, 'hir> {
     where
         F: FnOnce(&mut CheckLoopVisitor<'a, 'hir>),
     {
-        let old_cx = self.cx;
-        self.cx = cx;
+        self.cx_stack.push(cx);
         f(self);
-        self.cx = old_cx;
+        self.cx_stack.pop();
     }
 
-    fn require_break_cx(&self, name: &str, span: Span, break_span: Span) {
+    fn require_break_cx(&mut self, name: &str, span: Span, break_span: Span, cx_pos: usize) {
```

Comment:
Could `name` be a enum? It seems to only have a couple values?

Reply:
Done. Thank you.